XcodeAutoBasher/XcodeAutoBasher/Models/VOKProjectContainer.m

```diff
@@ -116,6 +116,52 @@ - (VOKDirectoryWatcher *)directoryWatcher
     return _directoryWatcher;
 }
 
+- (NSDictionary *)environmentVariables
+{
+    if ([NSThread isMainThread]) {
+        NSLog(@"VOKProjectContainer environmentVariables being retrieved on the main thread.");
+    }
+    NSPipe *outputPipe = [NSPipe pipe];
+    
+    NSTask *buildSettingsTask = [[NSTask alloc] init];
+    buildSettingsTask.launchPath = @"/usr/bin/xcodebuild";
+    buildSettingsTask.currentDirectoryPath = self.containingPath;
+    buildSettingsTask.arguments = @[ @"-showBuildSettings", ];
+    buildSettingsTask.standardOutput = outputPipe;
+    
+    [buildSettingsTask launch];
+    [buildSettingsTask waitUntilExit];
+    
+    NSData *outputData = [outputPipe.fileHandleForReading readDataToEndOfFile];
+    NSString *outputString = [[NSString alloc] initWithData:outputData encoding:NSUTF8StringEncoding];
+    
+    NSError *error;
+    NSRegularExpression *regex = [NSRegularExpression
+                                  regularExpressionWithPattern:
+                                  @"^[[:space:]]*([^[:space:]]*)[[:space:]]*=[[:space:]]*(.*)$"
+                                  options:NSRegularExpressionAnchorsMatchLines
+                                  error:&error];
+    if (!regex) {
+        NSLog(@"VOKProjectContainer error making regex: %@", error);
+        return nil;
+    }
+    
+    NSMutableDictionary *env = [NSMutableDictionary dictionary];
+    [regex
+     enumerateMatchesInString:outputString
+     options:NSMatchingReportProgress
+     range:NSMakeRange(0, outputString.length)
+     usingBlock:^(NSTextCheckingResult *result, NSMatchingFlags flags, BOOL *stop) {
+         if (result.numberOfRanges != 3) {
+             return;
+         }
+         id key = [outputString substringWithRange:[result rangeAtIndex:1]];
+         id value = [outputString substringWithRange:[result rangeAtIndex:2]];
```

Comment:
maybe "environmentVariable" instead of "key" - it's obvious we're using the env variable as a key, but it's not necessarily obvious that it is in fact the environment variable.